src/platform/plugins/shared/discover/public/application/main/discover_main_route.tsx

```diff
@@ -119,7 +119,7 @@ export const DiscoverMainRoute = ({
         initializeProfileDataViews(loadedRootProfileState).catch(() => {}),
       ]);
 
-      const userId = (await services.userProfile?.getCurrent())?.uid ?? '';
+      const userId = 'todo'; // TODO: fix getting the user id. Does not work with: (await services.userProfile?.getCurrent())?.uid ?? '';
```

Comment:
I also needed to get the userId for a PR that was not merged, but I will continue somewhere in the future, maybe this would work here?

` const currentUser = await services.core.security.authc.getCurrentUser();`

can be found here: 

https://github.com/elastic/kibana/pull/206894/files#r1955380830

CI didn't have any objections in this case

Reply:
@kertal Thanks for the suggestion! 
It does not throw exceptions anymore but for functional tests it returns no value.

Reply:
Oops, I suggested the same in a Slack thread without seeing @kertal already suggested it here 😁 This is the approach we use for field list caching, so could be reliable to use here as well. It's probably ok for it to be empty in tests where security isn't enabled. Maybe we can just use separate tests in x-pack or wherever security is enabled for testing this behaviour specifically?